File: problems/Minimum Window Substring/minWindow.cs
public class Solution {
    public string MinWindow(string s, string t) {
        var letterStore = new int[128];
        var result = "";

        foreach (var letter in t) {
            ++letterStore[letter];
        }

        for (int right = 0, left = 0, count = 0; s.Length > right; ++right) {
            if (0 < letterStore[s[right]]--) {
                ++count;
            }

            while (t.Length == count) {
                if ("" == result || result.Length > 1 + right - left) {
                    result = s.Substring(left, 1 + right - left);
                }

                if (0 < ++letterStore[s[left++]]) {
                   --count;
                }
            }
        }

        return result;
    }
}

File: problems/Minimum Window Substring/minWindow.java
class Solution {
    public String minWindow(String s, String t) {
        String result = "";
        int[] letterFrequencies = new int[128];

        for (char letter : t.toCharArray()) {
            ++letterFrequencies[letter - 'A'];
        }

        for (int left = 0, right = 0, count = 0; s.length() > right; ++right) {
            --letterFrequencies[s.charAt(right) - 'A'];

            if (0 <= letterFrequencies[s.charAt(right) - 'A']) {
                ++count;
            }

            while (t.length() == count) {
                if (result.isEmpty() || result.length() > 1 + right - left) {
                    result = s.substring(left, 1 + right);
                }

                ++letterFrequencies[s.charAt(left) - 'A'];
                
                if (0 < letterFrequencies[s.charAt(left) - 'A']) {
                    --count;
                }

                ++left;
            }
        }

        return result;
    }
}
